### PR TITLE
Create timeline events from Slack pins 

### PR DESCRIPTION
**Pull request by milesbxf**
Adds a TimelineEvent every time a Slack pin is added and removes it when
the pin is removed.

I also decided to migrate the metadata field from
`django.contrib.postgres.fields.JSONField` to `jsonfield.JSONField` -
this supports the same features in Postgres but additionally allows the
same support in other databases - crucially, in particular, SQLite,
which means we can write tests which interact with TimelineEvents.

### CODE REVIEW

File: response/core/models/timeline.py [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
-    EVENT_TYPES = (("text", "Freeform text field"),)
+class TimelineEvent(models.Model):
 
     incident = models.ForeignKey(Incident, on_delete=models.CASCADE)
     timestamp = models.DateTimeField(
         null=False, default=datetime.now, help_text="Time of when this event occurred."
     )
     text = models.TextField(help_text="Freeform text describing this event")
-    event_type = models.CharField(
-        max_length=10, choices=EVENT_TYPES, help_text="Type of event."
```
Comment (ChrisAnn) [inline response/core/models/timeline.py]: Do we no longer want to restrict it to this enum?

File: response/slack/models/pinned_message.py [hunk 2]
```diff
@@ -2,39 +2,66 @@
...
+    ExternalUser,
+    GetOrCreateSlackExternalUser,
+    TimelineEvent,
+)
 
 
 class PinnedMessageManager(models.Manager):
     def add_pin(self, incident, message_ts, author_id, text):
-        name = settings.SLACK_CLIENT.get_user_profile(author_id)['name']
+        name = settings.SLACK_CLIENT.get_user_profile(author_id)["name"]
```
Comment (ChrisAnn) [inline response/slack/models/pinned_message.py]: I've seen single quotes elsewhere.  Not that it's a blocker, I would like to know what's Python idiomatic :)

**Comment by danpalmer** [comment]
@milesbxf for my understanding, why would we want a timeline event to be removed? I can imagine pinned messages changing over time and that we wouldn't leave a message pinned after it is no longer relevant, but might want to see that on the timeline? Or is there another intended use-case here?

**Comment by milesbxf** [comment]
> @milesbxf for my understanding, why would we want a timeline event to be removed? I can imagine pinned messages changing over time and that we wouldn't leave a message pinned after it is no longer relevant, but might want to see that on the timeline? Or is there another intended use-case here?

@danpalmer good question. Internally we've found quite often in an incident we'll pin loads of messages as we go along to help us pick out info later. When we're compiling the report we often remove a lot of pinned messages, as they aren't generally written in a way that is easy to read (missing context) and add a lot of noise

File: response/core/models/timeline.py (hunk 1 above)
Comment (milesbxf) [reply response/core/models/timeline.py]: Nope. I realised that if people (including Monzo) want to add new custom event types to their private projects, they wouldn't be able to without forking this

**Comment by danpalmer** [comment]
@milesbxf Cool thanks for the explanation!

File: response/slack/models/pinned_message.py (hunk 2 above)
Comment (milesbxf) [reply response/slack/models/pinned_message.py]: Hah - unfortunately Python allows both, and both styles are widely used! Even the "official" Python style guide [PEP-8](https://www.python.org/dev/peps/pep-0008/#string-quotes) doesn't make a recommendation.

I use [black](https://github.com/psf/black) as a formatter locally - we should probably standardise on this

**Pull request merged**